File: lang/go/test/muse/algorithms/sorting/QuickSort/QuickSortTest.go
package main

import (
    "muse/algorithms/sorting/QuickSort"
    "muse/util/SequenceBuilder"
    "muse/util/Sequences"
    "muse/util/TestRunner"
)

func testQuickSort() bool {
    size := 32768

    arr := make([]int, size)
    SequenceBuilder.PackRandom(arr)

    checksum := Sequences.ParityChecksum(arr)

    QuickSort.Sort(arr)

    if Sequences.ParityChecksum(arr) != checksum {
        return false
    }

    return Sequences.IsSorted(arr)
}

func main() {
    TestRunner.ParseTest(testQuickSort())
}

File: ruby/test/muse/algorithms/sorting/QuickSortTest.rb
require "muse/algorithms/sorting/QuickSort"
require "muse/util/SequenceBuilder"
require "muse/util/Sequences"
require "muse/util/TestRunner"

def testQuickSort()
    size = 32768

    arr = Array.new(size)
    SequenceBuilder.packRandom(arr)

    checksum = Sequences.parityChecksum(arr)

    QuickSort.sort(arr)

    if Sequences.parityChecksum(arr) != checksum
        return false
    end

    Sequences.isSorted(arr)
end

if __FILE__ == $PROGRAM_NAME
    TestRunner.parseTest(testQuickSort())
end
